debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -100,6 +100,7 @@
     private final Clock clock;
     private final ElapsedTimeStrategy pollOutputDelay;
     private final EventProcessingFailureHandlingMode eventDeserializationFailureHandlingMode;
+    private final EventProcessingFailureHandlingMode eventStatementFailureHandlingMode;
```

Comment:
Do we really need a new option for this? Couldn't this also be controlled via `eventDeserializationFailureHandlingMode`?

Reply:
Hi, I will remake it, but I'll ask first: how much does mysql connector care about backward / forward compatibility of configs vs programs?
In case you don't care, it's better to merge the options and I'll do it. But if you do care about having compatible options (and unchanged behavior on upgrade), then we have to have another option to avoid changing features without announcing it first somehow.

Reply:
Good point; we do care about compatibility, _a lot_ :)

So the default behavior of that setting is `FAIL`, i.e. in that case there'd be no behavioral change. Only if someone had set it to `WARN` or `IGNORE`, they'd now not longer fail on unexpected "statement" style rows but move forward. Which seems ok-ish to me, but I also could understand someone finding this not acceptable.

@jpechane, @rk3rn3r, @Naros, any thoughts on this?

Reply:
@gunnarmorling IMHO this is fine as during the start of connector we check the global setting of `binlog_format`.
In fact this could be even improvement as we had reports in the past that people switch to row-based rpelication but Debezium still refused to start as there were still old statemenet-based replication messages in the binlog. They were then asked to flush the binlog etc. This could be easier solution for such scenario.

Reply:
@jpechane sorry I don't fully understand. So what is "this", that could be easier solution? Is it the original PR or is it related to the suggested changes?

Reply:
I am sorry for the confsion. IMHO we should just use the single `eventDeserializationFailureHandlingMode` for that purpose as proposed by @gunnarmorling . I don't think there is a real break in backward compatibility.

Reply:
Thanks, this should be fixed now.